stdlib/_zstd.pyi

```diff
@@ -0,0 +1,89 @@
+from _typeshed import ReadableBuffer
+from compression.zstd import _OptionsCompress, _OptionsDecompress
+from typing import Any, Final, Literal, final
+from typing_extensions import Self
+
+ZSTD_CLEVEL_DEFAULT: Final[int]
+ZSTD_DStreamOutSize: Final[int]
+ZSTD_btlazy2: Final[int]
+ZSTD_btopt: Final[int]
+ZSTD_btultra: Final[int]
+ZSTD_btultra2: Final[int]
+ZSTD_c_chainLog: Final[int]
+ZSTD_c_checksumFlag: Final[int]
+ZSTD_c_compressionLevel: Final[int]
+ZSTD_c_contentSizeFlag: Final[int]
+ZSTD_c_dictIDFlag: Final[int]
+ZSTD_c_enableLongDistanceMatching: Final[int]
+ZSTD_c_hashLog: Final[int]
+ZSTD_c_jobSize: Final[int]
+ZSTD_c_ldmBucketSizeLog: Final[int]
+ZSTD_c_ldmHashLog: Final[int]
+ZSTD_c_ldmHashRateLog: Final[int]
+ZSTD_c_ldmMinMatch: Final[int]
+ZSTD_c_minMatch: Final[int]
+ZSTD_c_nbWorkers: Final[int]
+ZSTD_c_overlapLog: Final[int]
+ZSTD_c_searchLog: Final[int]
+ZSTD_c_strategy: Final[int]
+ZSTD_c_targetLength: Final[int]
+ZSTD_c_windowLog: Final[int]
+ZSTD_d_windowLogMax: Final[int]
+ZSTD_dfast: Final[int]
+ZSTD_fast: Final[int]
+ZSTD_greedy: Final[int]
+ZSTD_lazy: Final[int]
+ZSTD_lazy2: Final[int]
+
+@final
+class ZstdCompressor:
+    CONTINUE: Final = 0
+    FLUSH_BLOCK: Final = 1
+    FLUSH_FRAME: Final = 2
+    def __init__(
+        self, level: int | None = ..., options: _OptionsCompress | None = ..., zstd_dict: ZstdDict | None = ...
+    ) -> None: ...
+    def compress(self, /, data: ReadableBuffer, mode: Literal[0, 1, 2] = ...) -> bytes: ...
```

Comment:
I think it might be useful to express these in terms of the `ZstdCompressor` attributes. It will make it easier to understand I think? If you think it makes the signature too busy then I suppose it is fine as is, but I want discourage passing bare integers.

Reply:
I believe the below is [invalid](https://typing.python.org/en/latest/spec/literal.html#legal-and-illegal-parameterizations):

```py
Literal[CONTINUE, FLUSH_BLOCK, FLUSH_FRAME]
```

But maybe I can try to do something like the following:

```py
_ZstdCompressorContinueType = Literal[0]
_ZstdCompressorFlushBlockType = Literal[1]
_ZstdCompressorFlushFrameType = Literal[2]

@final
class ZstdCompressor:
    CONTINUE: Final[_ZstdCompressorContinueType] = 0
    FLUSH_BLOCK: Final[_ZstdCompressorFlushBlockType] = 1
    FLUSH_FRAME: Final[_ZstdCompressorFlushFrameType] = 2
    def compress(self, /, data: ReadableBuffer, mode: _ZstdCompressorContinueType | _ZstdCompressorFlushBlockType | _ZstdCompressorFlushFrameType = ...) -> bytes: ...
```

Is it what you meant?

Reply:
Okay, I think that is fine then.